vertx-mongo-client/src/test/java/io/vertx/ext/mongo/MongoClientWithObjectIdTest.java

```diff
@@ -0,0 +1,158 @@
+package io.vertx.ext.mongo;
+
+import io.vertx.core.json.JsonObject;
+import org.bson.types.ObjectId;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+
+import static io.vertx.ext.mongo.WriteOption.ACKNOWLEDGED;
+
+public class MongoClientWithObjectIdTest extends MongoClientTestBase {
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    JsonObject config = getConfig();
+    config.put("useObjectId", true);
+    mongoClient = MongoClient.createNonShared(vertx, config);
+    CountDownLatch latch = new CountDownLatch(1);
+    dropCollections(latch);
+    awaitLatch(latch);
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    mongoClient.close();
+    super.tearDown();
+  }
+
+  protected void assertEquals(JsonObject expected, JsonObject actual) {
+
+    //Test cases will fail unless we map the $oid first
+    if (actual.containsKey("_id")) {
+      if (actual.getValue("_id") instanceof String) {
+        actual.put("_id", new JsonObject().put("$oid", actual.getString("_id")));
+      }
+    }
+    super.assertEquals(expected, actual);
+
+  }
+
+  @Test
+  @Override
+  public void testSavePreexistingLongID() throws Exception {
+    //Override this test as it does not make sense for useObjectId = true
+    assertTrue(true);
+    testComplete();
+    await();
+  }
```

Comment:
That overrides a test that doesn't make sense when useObjectId = true. A Long _id isn't valid